api/filter/accesslog.proto

```diff
@@ -9,24 +9,53 @@ import "google/protobuf/struct.proto";
 import "google/protobuf/timestamp.proto";
 import "google/protobuf/wrappers.proto";
 
-message AccessLogEntry {
-  // The HTTP request method (RFC 7231/2616)
-  RequestMethod request_method = 1;
+// Envoy access logs describe incoming interaction with Envoy over a fixed
+// period of time, and typically cover a single request/response exchange.
+// Except where explicitly declared otherwise, all fields describe
+// *downstream* interaction between Envoy and a connected client.
+// Fields describing *upstream* interaction are prefixed with the string
+// `upstream_`.
+message AccessLogBaseEntry {
+  // This field indicates the rate at which this log entry was sampled.
+  // Valid range is (0.0, 1.0].
+  double sample_rate = 1;
+
+  // This field is the IP and port on which the request from the user was
+  // received, stored in ipv4:port or [ipv6]:port format.
+  string destination_host = 2;
```

Comment:
I know this is just moving around what was there before, but I wonder if we want something more like https://github.com/envoyproxy/data-plane-api/blob/master/api/address.proto, i.e. structured, here. In general, in proto, it's better to avoid packing stuff into strings that can be expressed cleanly in a structured way.

Reply:
Sounds good to me, as long as it's still easy to populate the fields and we're not overloading any concepts. In general I agree it's preferable not to stuff data into strings that has an alternative, better representation.

Reply:
+1